### PR TITLE
Deep learning ner

### PR DESCRIPTION
**Pull request by aleksei-ai**
Deep Learning based NER

## Description
Added Tensorflow char-bilstm NER.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code improvements with no or little impact
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING](http://nlp.johnsnowlabs.com/contribute.html) page.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/scala/com/johnsnowlabs/util/FileHelper.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+import java.io.{File, IOException}
+import java.nio.charset.Charset
+import java.nio.file.{Files, Paths}
+
+import org.apache.commons.io.FileUtils
+
+object FileHelper {
+  def writeLines(file: String, lines: Seq[String], encoding: String = "UTF-8"): Unit = {
+    val writer = Files.newBufferedWriter(Paths.get(file), Charset.forName("UTF-8"))
```
[the remark this answers is not in the document]
Comment (aleksei-ai) [reply src/main/scala/com/johnsnowlabs/util/FileHelper.scala]: Will fix it. thank you

File: src/main/scala/com/johnsnowlabs/ml/tensorflow/DatasetEncoder.scala [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  /**
+    * Converts input data to format acceptable by Tensorflow Model
+    * @param sentences Batch of sentences. Every sentence - is array of words.
+    * @return Data that is acceptable by Tensorflow Model.
+    */
+  def encodeInputData(sentences: Array[Array[String]]): Batch = {
+    val batchSize = sentences.length
+    val sentenceLengths = sentences.map(s => s.length)
+
+    val maxSentenceLength = sentences.map(s => s.length).max
```
[the remark this answers is not in the document]
Comment (aleksei-ai) [reply src/main/scala/com/johnsnowlabs/ml/tensorflow/DatasetEncoder.scala]: Ok

File: src/main/scala/com/johnsnowlabs/nlp/annotators/ner/dl/NerDLApproach.scala [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+  def setTestDataset(er: ExternalResource) = set(testDataset, er)
+
+  override def verboseLevel = Verbose($(verbose))
+
+  setDefault(
+    minEpochs -> 0,
+    maxEpochs -> 50,
+    lr -> 0.2f,
+    po -> 0.05f,
```
[the remark this answers is not in the document]
Comment (aleksei-ai) [reply src/main/scala/com/johnsnowlabs/nlp/annotators/ner/dl/NerDLApproach.scala]: - Regarding decay - it's trade off between speed of train and quality. 0.05f - is seems as a standard good value.
- Regarding learning rate, short answer - is it's all goes from practice. But it's too long talk to explain it.

File: src/main/scala/com/johnsnowlabs/nlp/annotators/ner/dl/NerDLApproach.scala [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+    batchSize -> 9,
```
[the remark this answers is not in the document]
Comment (aleksei-ai) [reply src/main/scala/com/johnsnowlabs/nlp/annotators/ner/dl/NerDLApproach.scala]: Again it's quite standard.
